android-agent/src/main/java/io/opentelemetry/android/instrumentation/AndroidInstrumentation.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import android.app.Application
+import io.opentelemetry.android.OpenTelemetryRum
+
+interface AndroidInstrumentation {
+    companion object {
+        private val registry by lazy { AndroidInstrumentationRegistry.create() }
+
+        @JvmStatic
+        fun <T : AndroidInstrumentation> get(type: Class<out T>): T {
+            return registry.getByType(type)
+        }
+
+        @JvmStatic
+        fun getAll(): Collection<AndroidInstrumentation> {
+            return registry.getAll()
+        }
+    }
+
+    fun apply(
```

Comment:
I think `apply` is a little too vague/generic. To a novice, it could be unclear if `apply()` is/should-be called more than once, like maybe each time you call `apply()` it generates data or something.

I think a verb like `install()` could help things be a little clearer.